manifests/init.pp

```diff
@@ -130,12 +139,14 @@
       $svc_ensure  = $service_ensure
       $svc_enable  = $service_ensure ? { 'running' => true, 'stopped' => false }
       $file_ensure = 'present'
+      $srv_manage  = $manage_service
     }
     'absent', 'purged': {
       $pkg_ensure  = $ensure
       $svc_ensure  = 'stopped'
       $svc_enable  = false
       $file_ensure = 'absent'
+      $srv_manage  = false
```

Comment:
In Debian systems the `prerm` package would normally handle that. smartmontools is not an exception:

```
#!/bin/sh
set -e
# Automatically added by dh_installinit
if [ -x "/etc/init.d/smartmontools" ] || [ -e "/etc/init/smartmontools.conf" ]; then
        invoke-rc.d smartmontools stop || exit $?
fi
# End automatically added section
```

If you think about that, when you remove a package, whether it is in windows or linux, you also expect all the processes/services to have gone by the time the package is removed. If this doesn't happen, I would blame the package configuration scripts, not the configuration management tool.

I believe `$srv_manage` should be always equal to `false` when the package is absent. Otherwise how would you manage the service of an app that doesn't exist on the system?